src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -130,54 +123,105 @@ int IVsEditorFactory.Close()
                 case "Form":
 
                     // We must create the WinForms designer here
-                    var loaderName = GetWinFormsLoaderName(vsHierarchy);
-                    var designerService = (IVSMDDesignerService)_oleServiceProvider.QueryService<SVSMDDesignerService>();
-                    var designerLoader = (IVSMDDesignerLoader)designerService.CreateDesignerLoader(loaderName);
-                    if (designerLoader is null)
-                    {
-                        goto case "Code";
-                    }
+                    (pbstrEditorCaption, pguidCmdUI, ppunkDocView) = GetWinFormsDesignerData(vsHierarchy, itemid, textBuffer, readOnlyStatus, initializeView: true);
```

Comment:
Can we refactor this switch into a separate method? It's being used in GetEditorCaption too?

Reply:
Refactored the code to remove much of the churn.